*** src/lib/restrict-process-size.c.orig	2022-05-04 05:56:54.000000000 -0700
--- src/lib/restrict-process-size.c	2022-05-21 10:47:09.000000000 -0700
***************
*** 9,26 ****
  {
  	struct rlimit rlim;
  
! 	rlim.rlim_max = rlim.rlim_cur = bytes;
! 	if (setrlimit(RLIMIT_DATA, &rlim) < 0) {
! 		i_fatal("setrlimit(RLIMIT_DATA, %llu): %m",
! 			(unsigned long long)bytes);
! 	}
! 
! #ifdef HAVE_RLIMIT_AS
! 	if (setrlimit(RLIMIT_AS, &rlim) < 0) {
! 		i_fatal("setrlimit(RLIMIT_AS, %llu): %m",
! 			(unsigned long long)bytes);
! 	}
! #endif
  }
  
  void restrict_process_count(rlim_t count ATTR_UNUSED)
--- 9,31 ----
  {
  	struct rlimit rlim;
  
! /*
!  * Workaround for issue with dovercot version 2.3.17 on macOS 12
!  * https://www.mail-archive.com/dovecot@dovecot.org/msg84784.html
!  * 
!  *   rlim.rlim_max = rlim.rlim_cur = bytes;
!  *   if (setrlimit(RLIMIT_DATA, &rlim) < 0) {
!  *           i_fatal("setrlimit(RLIMIT_DATA, %llu): %m",
!  *                   (unsigned long long)bytes);
!  *   }
!  * 
!  * #ifdef HAVE_RLIMIT_AS
!  *   if (setrlimit(RLIMIT_AS, &rlim) < 0) {
!  *           i_fatal("setrlimit(RLIMIT_AS, %llu): %m",
!  *                   (unsigned long long)bytes);
!  *   }
!  * #endif
! */
  }
  
  void restrict_process_count(rlim_t count ATTR_UNUSED)
